apps/explorer/src/ui/RingChart.tsx

```diff
@@ -17,81 +19,97 @@ export interface RingChartProps {
+                        <div
+                            style={{ backgroundImage: color, color }}
+                            className="bg-clip-text text-body font-medium text-transparent"
+                        >
+                            {value} {label}
+                        </div>
                     </div>
                 ))}
             </div>
         </div>
     );
 }
 
-export function RingChart({
-    data,
-    radius = 20,
-    suffix,
-    title,
-}: RingChartProps) {
+export function RingChart({ data, radius = 20, title }: RingChartProps) {
     const cx = 25;
     const cy = 25;
     const dashArray = 2 * Math.PI * radius;
     const startAngle = -90;
     const total = data.reduce((acc, { value }) => acc + value, 0);
     let filled = 0;
-    const segments = data.map(({ value, label, color }) => {
+    const segments = data.map(({ value, label, color, gradient }) => {
         const ratio = (100 / total) * value;
         const angle = (filled * 360) / 100 + startAngle;
         const offset = dashArray - (dashArray * ratio) / 100;
         filled += ratio;
         return (
-            <circle
-                key={label}
-                cx={cx}
-                cy={cy}
-                r={radius}
-                fill="transparent"
-                stroke={color}
-                strokeWidth={4}
-                strokeDasharray={dashArray}
-                strokeDashoffset={offset}
-                transform={`rotate(${angle} ${cx} ${cy})`}
-            />
+            <>
+                {gradient && (
+                    <defs>
+                        <linearGradient id="gradient">
+                            {gradient.map(({ percent, color }, i) => (
+                                <stop
+                                    key={i}
+                                    offset={percent}
+                                    stopColor={color}
+                                />
+                            ))}
+                        </linearGradient>
+                    </defs>
+                )}
+                <circle
+                    key={label}
+                    cx={cx}
+                    cy={cy}
+                    r={radius}
+                    fill="transparent"
+                    stroke={gradient ? 'url(#gradient)' : color}
+                    strokeWidth={5}
+                    strokeDasharray={dashArray}
+                    strokeDashoffset={offset}
+                    transform={`rotate(${angle} ${cx} ${cy})`}
+                />
+            </>
         );
     });
 
     return (
-        <div className="grid grid-flow-row items-center justify-start xl:grid-cols-2 xl:justify-center">
-            <div className="self-start">
-                <Legend data={data} title={title} />
-            </div>
-
-            <div className="relative">
+        <div className="flex items-center gap-5">
+            <div className="relative h-24 min-h-[96px] w-24 min-w-[96px]">
```

Comment:
Why an explicit height/width and also a min one?

Reply:
Also I never love explicit heights and widths, especially in a base component like this. We either should make this more generic or we should hoist the entire component out of here and into the relevant UI surface